arangod/VocBase/Methods/UpgradeTasks.cpp

```diff
@@ -317,40 +317,23 @@ Result createSystemStatisticsCollections(
     };
     std::vector<std::shared_ptr<VPackBuffer<uint8_t>>> buffers;
     Result res;
+    OperationOptions options(ExecContext::current());
     for (auto const& collection : systemCollections) {
+      // No need to batch this.
+      // Fresh databases will have a batch run for those collections already.
+      // We only hit this on databases that do not have statistics collections
+      // yet. Which have to be somewhere from the 2.X series, and never had an
+      // upgrade task.
       std::shared_ptr<LogicalCollection> col;
-      res = methods::Collections::lookup(vocbase, collection, col);
-      if (col) {
-        createdCollections.emplace_back(std::move(col));
-      }
-      if (res.is(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND)) {
-        // if not found, create it
-        VPackBuilder options;
-        options.openObject();
-        options.add(StaticStrings::DataSourceSystem, VPackSlice::trueSlice());
-        options.add(StaticStrings::WaitForSyncString, VPackSlice::falseSlice());
-        options.close();
-
-        systemCollectionsToCreate.emplace_back(CollectionCreationInfo{
-            collection, TRI_COL_TYPE_DOCUMENT, options.slice()});
-        buffers.emplace_back(options.steal());
-      }
-    }
-
-    // We capture the vector of created LogicalCollections here
-    // to use it to create indices later.
-    if (!systemCollectionsToCreate.empty()) {
-      std::vector<std::shared_ptr<LogicalCollection>> cols;
-      OperationOptions options(ExecContext::current());
-      res = methods::Collections::create(
-          vocbase, options, systemCollectionsToCreate, true, false, false,
-          nullptr, cols, true /* allow system collection creation */);
+      res = methods::Collections::createSystem(vocbase, options, collection,
+                                               false, col);
       if (res.fail()) {
         return res;
       }
-      // capture created collection vector
-      createdCollections.insert(std::end(createdCollections), std::begin(cols),
-                                std::end(cols));
+      TRI_ASSERT(col) << "Create system collection did not fail but also did "
+                         "not create a collection.";
+      createdCollections.emplace_back(std::move(col));
+      res = methods::Collections::lookup(vocbase, collection, col);
```

Comment:
This is a use-after-move for `col`.

Reply:
😲

Reply:
Fixed, the lookup was not necessary anyhow.